### PR TITLE
ENH: Added pod configuration for fio

### LINKED ISSUE #51: Add affinity related structs
https://github.com/xridge/kubestone/blob/a24f5d1a70d3b195f01067cbe3d0675f901e2a5c/api/v1alpha1/fio_types.go#L39-L44

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in a24f5d1a70d3b195f01067cbe3d0675f901e2a5c when #46 was merged. cc @julio-santiesteban.

### PR DESCRIPTION
**Pull request by julio-santiesteban**
Fixes #51

### CODE REVIEW

File: api/v1alpha1/fio_types.go [hunk 1]
```diff
@@ -20,6 +20,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// FioPodSpec contains the configuration for the benchmark pods
+type FioPodConfigurationSpec struct {
```
Comment (otto-von-bivouac) [inline api/v1alpha1/fio_types.go]: Currently this deviates from iperf3 controller, there we have `PodLabels` and `PodScheduling` in the top level Spec.
I'm not against moving this under a common spec, but we should do this consistently throughout the whole project.

Now, given that we should also expose CPU and Memory Limits (and maybe others), it is a good idea to have a common struct as you propose. 

Can you please move this struct to `k8s_types.go` in api/ ?

CC: @otto-von-bivouac

Comment (julio-santiesteban) [reply api/v1alpha1/fio_types.go]: ?
On the contrary, I'm using the same structure as in iperf3: https://github.com/xridge/kubestone/blob/258970513f9e89687c9ee0921c43b5699fbd0c47/api/v1alpha1/iperf3_types.go#L26

However, I do agree that we should move these to k8s_types. The problem is, the two benchmarks need slightly different structures (that's why I kept it separated for now). Maybe do the refactor in a follow up PR?

Comment (otto-von-bivouac) [reply api/v1alpha1/fio_types.go]: I think we should only factor the common ones: `PodScheduling`, `PodLabels`. 
In other words, Iperf3ConfigurationSpec should also use a `PodConfigurationSpec`, which contains those fields.

I'm okay with a follow-up PR, but then please add a TODO line for it.

Comment (julio-santiesteban) [reply api/v1alpha1/fio_types.go]: Nvm, I did it in 5a3284d.

Comment (otto-von-bivouac) [reply api/v1alpha1/fio_types.go]: thanks. it is still differing on API level (inlined object in iperf3), but I'll address it in a separate PR.

**Pull request merged**